libs/sdk-ui-pivot/src/impl/structure/tableDescriptorFactory.ts

```diff
@@ -246,21 +247,30 @@ function createColumnDescriptorsFromDataSeries(dv: DataViewFacade): GroupingOper
     return groupColumns(leafColumns, scopingAttributes);
 }
 
+function createRowDescriptor(index: number, attributeDescriptor: IAttributeDescriptor): SliceCol {
+    return {
+        type: "sliceCol",
+        id: `r_${index}`,
+        index: index,
+        attributeDescriptor,
+        fullIndexPathToHere: [index],
+        effectiveTotals: attributeDescriptor.attributeHeader.totalItems ?? [],
+    };
+}
+
 function createRowDescriptors(dv: DataViewFacade): SliceCol[] {
-    return dv
-        .data()
-        .slices()
-        .descriptors.filter(isAttributeDescriptor)
-        .map((attributeDescriptor, idx) => {
-            return {
-                type: "sliceCol",
-                id: `r_${idx}`,
-                index: idx,
-                attributeDescriptor,
-                fullIndexPathToHere: [idx],
-                effectiveTotals: attributeDescriptor.attributeHeader.totalItems ?? [],
-            };
-        });
+    if (getMeasureGroupDimensionIndex(dv) === 0) {
+        return dv
+            .meta()
+            .attributeDescriptorsForDim(0)
+            .map((attributeDescriptor, idx) => createRowDescriptor(idx, attributeDescriptor));
```

Comment:
This is a nitpick, but if you would flip the signature of createRowDescriptor to be like (attributeDescriptor, idx), you could then instead of 
```
.map((attributeDescriptor, idx) => createRowDescriptor(idx, attributeDescriptor))
```
use just
```
.map(createRowDescriptor)
```

Reply:
fixed